File: Python/minimum-index-sum-of-two-lists.py
class Solution:
    def findRestaurant(self, list1, list2):
        """
        :type list1: List[str]
        :type list2: List[str]
        :rtype: List[str]
        """
        lookup = {}
        for i, s in enumerate(list1):
            lookup[s] = i

        res = []
        minSum = float("inf")

        for j, s in enumerate(list2):
            if s in lookup:
                if j + lookup[s] < minSum:
                    res = [s]
                    minSum = j + lookup[s]
                elif j + lookup[s] == minSum:
                    res.append(s)

        return res

File: Swift/minimum-index-sum-of-two-lists.swift
class Solution {
    func findRestaurant(_ list1: [String], _ list2: [String]) -> [String] {
        var lookup = [String: Int]()
        for (i, str) in list1.enumerated() {
            lookup[str] = i
        }

        var res = [String]()
        var minSum = Int.max

        for (j, str) in list2.enumerated() {
            if let i = lookup[str] {
                if i + j < minSum {
                    res = [str]
                    minSum = i + j
                } else if i + j == minSum {
                    res.append(str)
                }
            }
        }

        return res
    }
}
